scvi/external/cell2location/autoguide.py

```diff
@@ -0,0 +1,384 @@
+            # init_param = torch.normal(torch.full(size=param_dim, fill_value=0.),
+            #                          torch.full(param_dim, 1. / np.sqrt(self.n_hidden)),
+            #                          device=site["value"].device)
+            init_param = np.random.normal(
+                np.zeros(param_dim),
+                (np.ones(param_dim) * self.init_param_scale) / np.sqrt(self.n_hidden),
+            ).astype("float32")
+            _deep_setattr(
+                self.hidden2locs,
+                name,
+                PyroParam(
+                    torch.tensor(
+                        init_param, device=site["value"].device, requires_grad=True
+                    )
+                ),
+            )
+
+            init_param = np.random.normal(
+                np.zeros(param_dim),
+                (np.ones(param_dim) * self.init_param_scale) / np.sqrt(self.n_hidden),
+            ).astype("float32")
+            _deep_setattr(
+                self.hidden2scales,
+                name,
+                PyroParam(
+                    torch.tensor(
+                        init_param, device=site["value"].device, requires_grad=True
+                    )
+                ),
+            )
+
+            if not self.single_encoder:
+                _deep_setattr(
+                    self.encoder,
+                    name,
+                    self.encoder_class(
+                        n_in=self.n_in, n_out=self.n_hidden, **self.encoder_kwargs
+                    ).to(site["value"].device),
+                )
+
+    def _get_loc_and_scale(self, name, encoded_hidden):
+        """
+        Get mean (loc) and sd (scale) of the posterior distribution, as a linear function of encoder hidden layer.
+        Parameters
+        ----------
+        name
+            variable name
+        encoded_hidden
+            tensor when `single_encoder==True` and dictionary of tensors for each site when `single_encoder=False`
+
+        """
+
+        linear_locs = _deep_getattr(self.hidden2locs, name)
+        linear_scales = _deep_getattr(self.hidden2scales, name)
+
+        if not self.single_encoder:
+            # when using multiple encoders extract hidden layer for this parameter
+            encoded_hidden = encoded_hidden[name]
+
+        locs = encoded_hidden @ linear_locs
+        scales = self.softplus((encoded_hidden @ linear_scales) - 2)
+
+        return locs, scales
+
+    def encode(self, *args, **kwargs):
+        """
+        Apply encoder network to input data to obtain hidden layer encoding.
+        Parameters
+        ----------
+        args
+            Pyro model args
+        kwargs
+            Pyro model kwargs
+        -------
+
+        """
+        in_names = self.amortised_plate_sites["in"]
+        x_in = [kwargs[i] if i in kwargs.keys() else args[i] for i in in_names]
+        # apply data_transform
+        x_in = [self.data_transform(x) for x in x_in]
```

Comment:
support `data_transform==None` for identity?

Reply:
Default value could also be `data_transform=lambda x: x`.
Log-transform is good default in the context of scVI.